source/AI.cpp

```diff
@@ -1087,20 +1087,20 @@ bool AI::HasHelper(const Ship &ship, const bool needsFuel)
 
 
 // Pick a new target for the given ship.
-shared_ptr<Ship> AI::FindTarget(const Ship &ship) const
+void AI::FindTarget(Ship &ship) const
 {
 	// If this ship has no government, it has no enemies.
 	shared_ptr<Ship> target;
 	const Government *gov = ship.GetGovernment();
 	if(!gov || ship.GetPersonality().IsPacifist())
-		return target;
+		return ship.SetTargetShip(target);
```

Comment:
This is a bit of a confusing way to return void. Can we just have the `ship.SetTargetShip(...)` and the return statements on different lines?

Reply:
Well the function can only return void anyway this is mostly a way to end the function

Reply:
Sure I'll do it